test/entities/swapV2.integration.test.ts

```diff
@@ -0,0 +1,210 @@
+
+const { rpcUrl } = await startFork(ANVIL_NETWORKS.MAINNET, undefined, blockNo);
+
+const BAL = TOKENS[chainId].BAL;
+const WETH = TOKENS[chainId].WETH;
+
+describe('SwapV2', () => {
+    let client: Client & PublicActions & TestActions & WalletActions;
+    let testAddress: Address;
+    const pathBalWeth: Path = {
+        balancerVersion: 2,
+        tokens: [
+            {
+                address: TOKENS[chainId].BAL.address,
+                decimals: TOKENS[chainId].BAL.decimals,
+            },
+            {
+                address: TOKENS[chainId].WETH.address,
+                decimals: TOKENS[chainId].WETH.decimals,
+            },
+        ],
+        pools: [
+            '0x5c6ee304399dbdb9c8ef030ab642b10820db8f56000200000000000000000014',
+        ],
+        inputAmountRaw: 100000000000n,
+        outputAmountRaw: 100000000000n,
+    };
+
+    beforeAll(async () => {
+        client = createTestClient({
+            mode: 'anvil',
+            chain: CHAINS[chainId],
+            transport: http(rpcUrl),
+        })
+            .extend(publicActions)
+            .extend(walletActions);
+
+        testAddress = (await client.getAddresses())[0];
+    });
+
+    beforeEach(async () => {
+        await forkSetup(
+            client,
+            testAddress,
+            [WETH.address, BAL.address],
+            [WETH.slot as number, BAL.slot as number],
+            [parseEther('100'), parseEther('100')],
+            undefined,
+            balancerVersion,
+        );
+    });
+
+    describe('query method should return correct updated', () => {
+        test('GivenIn', async () => {
+            const swap = new SwapV2({
+                chainId,
+                paths: [pathBalWeth],
+                swapKind: SwapKind.GivenIn,
+            });
+
+            const updated = await swap.query(rpcUrl);
+
+            const wethToken = new Token(
+                chainId,
+                TOKENS[chainId].WETH.address,
+                TOKENS[chainId].WETH.decimals,
+            );
+            expect(updated.token).to.deep.eq(wethToken);
+            expect(updated.amount).to.eq(44236888n);
+        });
+        test('GivenOut', async () => {
+            const swap = new SwapV2({
+                chainId,
+                paths: [pathBalWeth],
+                swapKind: SwapKind.GivenOut,
+            });
+
+            const updated = await swap.query(rpcUrl);
+
+            const balToken = new Token(
+                chainId,
+                TOKENS[chainId].BAL.address,
+                TOKENS[chainId].BAL.decimals,
+            );
+            expect(updated.token).to.deep.eq(balToken);
+            expect(updated.amount).to.eq(60635225778147n);
+        });
+    });
+    describe('swap should be executed correcly', () => {
+        describe('wethIsEth: false', () => {
+            const swapParams = {
+                chainId,
+                paths: [pathBalWeth],
+                wethIsEth: false,
```

Comment:
Swap doesn't expect wethIsEth as input, right? Is this actually being applied somewhere?

Reply:
It doesn't. I must have missed this during the refactor. Removed it now.